lib/matplotlib/gridspec.py

```diff
@@ -202,6 +202,31 @@ def get_grid_positions(self, fig, raw=False):
         fig_lefts, fig_rights = (left + cell_ws).reshape((-1, 2)).T
         return fig_bottoms, fig_tops, fig_lefts, fig_rights
 
+    @staticmethod
+    def _check_gridspec_exists(figure, nrows, ncols):
+        """
+        Check if the figure already has a gridspec with these dimensions,
+        or create a new one
+        """
+        gs = None
+        for ax in figure.get_axes():
+            if hasattr(ax, 'get_subplotspec'):
+                ggs = ax.get_subplotspec().get_gridspec()
+                if hasattr(ggs, 'get_topmost_subplotspec'):
+                    # This is needed for colorbar gridspec layouts.
+                    # This is probably OK becase this whole logic tree
+                    # is for when the user is doing simple things with the
+                    # add_subplot command.  Complicated stuff, the proper
+                    # gridspec is passed in...
```

Comment:
perhaps further add a comment that this is only for cases like subplot(ijk) where the user cannot be explicitly passing in e.g. width_ratios or height_ratios, so we don't need to try to match these.

Reply:
Clarified to 
```
# For complicated layouts like subgridspecs the proper gridspec is passed in...
```
I think width_ratios etc are not the problem, but nested gridspecs are not particularly handled here.  But I don't know how you add a subplot to a nested gridspec without grabbing its subplotspec, so I think this is fine.